app/assets/javascripts/discourse/app/lib/plugin-api.js

```diff
@@ -1981,6 +1982,35 @@ class PluginApi {
   registerModelTransformer(modelName, transformer) {
     registerModelTransformer(modelName, transformer);
   }
+
+  /**
+   * EXPERIMENTAL. Do not use.
+   *
+   * When initiating a search inside the composer or other designated inputs
+   * with the `#` key, we search records based on params registered with
+   * this function, and order them by type using the priority here. Since
+   * there can be many different inputs that use `#` and some may need to
+   * weight different types higher in priority, we also require a context
+   * parameter.
+   *
+   * For example, the topic composer may wish to search for categories
+   * and tags, with categories appearing first in the results. The usage
+   * looks like this:
+   *
+   * api.registerHashtagSearchParam("category", "topic-composer", 100);
+   * api.registerHashtagSearchParam("tag", "topic-composer", 50);
+   *
+   * Additional types of records used for the hashtag search results
+   * can be registered via the #register_hashtag_data_source plugin API
+   * method.
+   *
+   * @param {string} param - The type of record to be fetched.
+   * @param {string} context - Where the hashtag search is being initiated using `#`
+   * @param {number} priority - Used for ordering types of records. Priority order is descending.
+   */
+  registerHashtagSearchParam(param, context, priority) {
+    registerHashtagSearchParam(param, context, priority);
+  }
 }
```

Comment:
Should we bump api version?

Reply:
I don’t want to bump it yet since it’s experimental I’ll do that once it’s more solid